src/main/java/com/aws/greengrass/logmanager/LogManagerService.java

```diff
@@ -628,6 +648,55 @@ private void processLogsAndUpload() throws InterruptedException {
         }
     }
 
+    /**
+     * calculate the hash for the active file, which takes and digests first some lines (default is 1) in the file.
+     * If the content of file is insufficient, or incorrect setting for DEFAULT_LINES_FOR_DIGEST_NUM, the hash is null.
+     * @param activeFile
+     * @return the nullable calculated hash for active file
+     */
+    private Optional<String> calculateHashForActiveFile(File activeFile) {
+        Optional activeFileHash = Optional.ofNullable(null);
+        // String activeFileHash = null;
+        try (SeekableByteChannel chan = Files.newByteChannel(activeFile.toPath(), StandardOpenOption.READ);
+             PositionTrackingBufferedReader r =
+                new PositionTrackingBufferedReader(new InputStreamReader(Channels.newInputStream(chan),
+                        StandardCharsets.UTF_8))) {
```

Comment:
you don't need any of this complication for reading the file. Just use `Files.newBufferedReader`